exploration/maintaining-registry.md

```diff
@@ -0,0 +1,114 @@
+Our goal is to be as universal as possible, 
+making MFv2's message syntax available to developers in many different
+runtimes in a wholly consistent manner.
+Because we want broad adoption in many different programming environments
+and because the capabilities 
+and functionality available in these environments varies widely,
+the default function registry must be conservative in what it required.
+
+At the same time, we want to promote message interoperability.
+Even though a feature cannot be adopted by all platforms,
+diversity in the function names, operands, options, error behavior,
+and so forth remains undesirable.
+This suggests that there exist a registry at a lower level of normativeness
+with "templates" for functions that implementations can implement 
+when they have the wherewithal to support them.
+
+An example of this might be "personal name formatting",
+whose functionality is new in CLDR and ICU.
+Few if any non-ICU implementations are currently prepared to implement
+a function such as `:person`
+and implementation and usage experience is limited in ICU.
+Where functionality is made available, we don't want it to vary from
+platform to platform.
+
+Finally, we need to establish mechanisms for managing proposals
+for the various types of function registry.
+
+## Use-Cases
+
+_What use-cases do we see? Ideally, quote concrete examples._
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+To address the above problem, in addition to the default registry,
+create a template for registry functions
+and a process for proposing and evaluating these functions for inclusion
+into the MessageFormat v2 function registry.
+
+There would be three levels of expected maturity:
+
+- **Default Registry** includes functions that are normatively required by all implementations.
+  Such entries will be limited in scope to functions that can be
+  implemented in nearly any programming environment.
+  > Examples: `:string`, `:number`, `:date`
+- **Recommended for General Implementation**
+  ("RGI", deliberately similar to RGI in emoji, tho' we may want to change the name)
+  includes functions that are not
+  normatively required but whose names, operands, and options are recommended.
+  Implementations are _strongly_ encouraged to use these function signatures
+  when implementing the described functionality.
+  This will promote message interoperability
+  and reduce the learning curve for developers, tools, and translators.
+  > Examples: We don't currently have any, but potential work here
+  > might includes personal name formatting, gender-based selectors, etc.
+- **Unicode Extensions** includes optional functionality that implementations
```

Comment:
I get the RGI bucket of definitions, but I'm having trouble seeing what use we'd make of the `u` space. If it's just for stuff that we'll likely standardise in the anon namespace later on, I think some other namespace name would be better for that, like `x` or `next`. Or is it meant for something different?

Reply:
One example is something @sffc mentioned. Suppose we wanted to define date skeletons or number skeletons in a referenceable way. We don't want to add them to `:datetime` or `:number`, because then everyone would have to implement them. But we could define them as `:u:skeleton` options which implementers SHOULD follow if they want to provide this feature.

Similarly, this provides a way to prototype functionality before making it RGI.

We can debate spelling of it. I chose `u` for familiarity.

Reply:
I'd prefer the prototyping to use the same name as what we ultimately intend for the thing, be it a function or an option. We could classify each defined thing according to some progressive schema like alpha/beta/stable, and only promise that the "stable" stuff will actually be stable.